src/rules/effects/use-action-creator-in-of-type.ts

```diff
@@ -0,0 +1,36 @@
+import type { TSESTree } from '@typescript-eslint/experimental-utils'
+import { ESLintUtils } from '@typescript-eslint/experimental-utils'
+import path from 'path'
+import { docsUrl } from '../../utils'
+
+export const messageId = 'noOfTypeLiteral'
+export type MessageIds = typeof messageId
+
+type Options = []
+
+export default ESLintUtils.RuleCreator(docsUrl)<Options, MessageIds>({
+  name: path.parse(__filename).name,
```

Comment:
I went with `use-action-creator-in-of-type` to conform to the `use-action-creator-in-dispatch` as per discussion in https://github.com/timdeschryver/eslint-plugin-ngrx/issues/238#issuecomment-931360802.

Reply:
After thinking twice, I realized that `prefer-*` seems to be fits better in this context and the other in #249... so I'd prefer `prefer-action-creator-in-of-type` and `prefer-action-creator-in-dispatch`.

---

From here on are just random thoughts/suggestions that involve breaking changes and that don't involve this PR...

About the existing rules with the prefix `use-*` (`use-selector-in-select` and `use-consistent-global-store`), I'd suggest to rename these to `prefer-selector-in-select` (for the same reason mentioned above) and `consistent-global-store`(it seems that starting with `consistent-*` is a standard  for this kind of rule... check the `consistent-*` rules [here](https://github.com/eslint/eslint/tree/master/lib/rules) and [here](https://github.com/typescript-eslint/typescript-eslint/tree/master/packages/eslint-plugin/src/rules)).

On other hand, I guess the `use-effects-lifecycle-interface` makes sense to keep as is.

Reply:
+1 for `prefer`.
The suggestion to rename the existing rule is indeed better.
I'll create a new issue; we can do it later when we've stabilized.
Maybe that other breaking changes will come to light.